pkg/cache/v3/linear.go

```diff
@@ -239,6 +239,45 @@ func (cache *LinearCache) CreateWatch(request *Request) (chan Response, func())
 	}
 }
 
+// SetResources replaces current resources with a new set of resources
+//
+// If we have individual watches for two resources (there are 2 DiscoveryRequest with resourceNames=res1 and resourceNames=res2)
+// Then if you set new resources, res1 changed but res2 is not, we assume that all resources are changed so both watches will be triggered.
+// That's why this function is well suited for a use case when resourceNames of DiscoveryRequest is empty.
+// This way you don't have to call UpdateResource for every resource which will trigger watch of DiscoveryRequest(resourceNames="") twice,
+// but it will only happen once regardless of how many resources are changed.
```

Comment:
I think this comment might be a little much for this function. If I understand this correctly, this is a good hook to use wildcarding when requesting resources (resourceName being empty), maybe we should just allude to that here instead?

Reply:
I rewrote the comment. Please take a look if it's ok, if not I could use some guidance. Thanks!